lib/telephony/otp_sender.rb

```diff
@@ -10,13 +10,13 @@ def initialize(to:, otp:, expiration:, channel:)
     end
 
     def send_authentication_otp
-      response = adapter.send(message: authentication_message, to: recipient_phone)
+      response = adapter.send(message: authentication_message, to: recipient_phone, otp: otp)
```

Comment:
Do we need to pass in `otp` to the adapter? One thing that stands out to me is if we care for the gem to be broadly useful for SMS/voice generally, vs. tailored to OTP. It'd seem reasonable to have an `otp` attribute in this class, though I don't know that the `VoiceSender` and `SMSSender` classes need to (or should need to) have the attribute as well.

Reply:
The reason I made OTP a first-class field was because our code to extract OTPs via regex was pretty brittle, is used a lot by the IDP test suite: https://github.com/18F/identity-telephony/pull/45/commits/ec3baa4ac1ac355d6f88525eecfb66a93cc2c752#diff-85abe22d52c8b558ca5d52768533c3c702aa36eb7a5ba6d1c6ec5b25d8832519L30-L33

Right now, our gem is very tailored to OTP, it maintains the translation strings for the messages we send, so I don't think it's too unreasonable to pass this as a field

Reply:
Mm, okay, I agree it makes sense in `OtpSender`, I guess I don't really follow why all the voice / SMS tests need to be so concerned with the OTP. But since that's how it exists today, seems fine or at least consistent.

Reply:
So the IDP has a helper method that's called in like 90 places, `fill_in_code_with_last_phone_otp`:

https://github.com/18F/identity-idp/blob/38b80b5df4b1678fac9c2d1b15463699e56c78e9/spec/support/features/session_helper.rb#L263-L265

and it uses these helper methods to read the OTP back off the adapters:

https://github.com/18F/identity-idp/blob/38b80b5df4b1678fac9c2d1b15463699e56c78e9/spec/support/otp_helper.rb#L10-L18

Reply:
Gotcha, that makes more sense. Thanks 👍